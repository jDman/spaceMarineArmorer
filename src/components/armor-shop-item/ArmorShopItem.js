import React from 'react';

import Input from '../form/input/Input';
import Button from '../button/Button';

import classes from './ArmorShopItem.module.css';

const ArmorShopItem = (props) => {
  const armor = props.armor;
  return (
    <li className={classes.ArmorShopListItem}>
      <div className={classes.ArmorShopListItemContainer}>
        <p>
          <strong>Name</strong>: {armor.name}
        </p>
        <p>
          <strong>Type</strong>: {armor.type}
        </p>
        <p>
          <strong>Description</strong>: {armor.description}
        </p>
        <p>
          <strong>Company</strong>: {armor.company.trim().split('_').join(' ')}
        </p>
        <p>
          <strong>Quality</strong>: {armor.quality}
        </p>
        <p>
          <strong>Protection</strong>: {armor.protection}
        </p>
        <p>
          <strong>Shield</strong>: {armor.shield}
        </p>
        <p>
          <strong>Discount</strong>: {armor.discount}
        </p>

        <p>
          <strong>Cost</strong>: {armor.cost}
        </p>

        <small>Created By: {armor.createdBy}</small>
      </div>

      <div className={classes.ArmorShopListItemContainer}>
        <form
          onSubmit={(event) => props.addToCart(event, armor)}
          className={classes.ArmorShopListForm}
        >
          <Input
            elementType={armor.config.elementType}
            elementConfig={armor.config.elementConfig}
            classes={classes.ArmorShopListFormElement}
            value={armor.config.value}
            invalid={!armor.config.valid}
            touched={armor.config.touched}
            disabled={armor.stock < 1}
            hasValidation={armor.config.validation}
            changed={(event) =>
              props.inputChangedHandler(event.target.value, armor.id)
            }
          />

          <Button
            type="submit"
            classes={
              armor.stock > 0
                ? classes.ArmorShopAddToCartButton
                : [
                    classes.ArmorShopAddToCartButton,
                    classes.ArmorShopAddToCartButtonDisabled,
                  ]
            }
            disabled={!armor.config.valid}
            buttonText={armor.stock > 0 ? 'Add to Cart' : 'Out of Stock'}
          />
        </form>
      </div>
    </li>
  );
};

export default ArmorShopItem;
