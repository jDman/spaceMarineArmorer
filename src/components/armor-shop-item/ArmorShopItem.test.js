import React from 'react';
import renderer from 'react-test-renderer';

import ArmorShopItem from './ArmorShopItem';

describe('<ArmorShopItem />', () => {
  it('should render with props passed in', () => {
    const armor = {
      id: '123',
      name: 'Legatron',
      type: 'leg',
      description: 'Leg Armor that is basic',
      company: 'orian_labs',
      quality: 'low',
      protection: 'low',
      shield: 0,
      discount: 0,
      cost: 100,
      stock: 0,
      createdBy: 'Freddy',
      config: {
        elementType: 'input',
        elementConfig: {
          type: 'number',
          name: 'quantity',
          min: 0,
          max: 0,
          step: 1,
        },
        value: 0,
        valid: false,
        validators: [() => {}],
        touched: false,
      },
    };

    const component = renderer.create(
      <ArmorShopItem
        armor={armor}
        inputChangedHandler={() => {}}
        addToCart={() => {}}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render Add to Cart button when stock available', () => {
    const armor = {
      id: '123',
      name: 'Legatron',
      type: 'leg',
      description: 'Leg Armor that is basic',
      company: 'orian_labs',
      quality: 'low',
      protection: 'low',
      shield: 0,
      discount: 0,
      cost: 100,
      stock: 1,
      createdBy: 'Freddy',
      config: {
        elementType: 'input',
        elementConfig: {
          type: 'number',
          name: 'quantity',
          min: 0,
          max: 1,
          step: 1,
        },
        value: 1,
        valid: true,
        validators: [() => true],
        touched: true,
      },
    };

    const component = renderer.create(
      <ArmorShopItem
        armor={armor}
        inputChangedHandler={() => {}}
        addToCart={() => {}}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
