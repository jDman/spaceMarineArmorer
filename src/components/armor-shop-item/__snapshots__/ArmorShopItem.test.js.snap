// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArmorShopItem /> should render Add to Cart button when stock available 1`] = `
<li
  className="ArmorShopListItem"
>
  <div
    className="ArmorShopListItemContainer"
  >
    <p>
      <strong>
        Name
      </strong>
      : 
      Legatron
    </p>
    <p>
      <strong>
        Type
      </strong>
      : 
      leg
    </p>
    <p>
      <strong>
        Description
      </strong>
      : 
      Leg Armor that is basic
    </p>
    <p>
      <strong>
        Company
      </strong>
      : 
      orian labs
    </p>
    <p>
      <strong>
        Quality
      </strong>
      : 
      low
    </p>
    <p>
      <strong>
        Protection
      </strong>
      : 
      low
    </p>
    <p>
      <strong>
        Shield
      </strong>
      : 
      0
    </p>
    <p>
      <strong>
        Discount
      </strong>
      : 
      0
    </p>
    <p>
      <strong>
        Cost
      </strong>
      : 
      100
    </p>
    <small>
      Created By: 
      Freddy
    </small>
  </div>
  <div
    className="ArmorShopListItemContainer"
  >
    <form
      className="ArmorShopListForm"
      onSubmit={[Function]}
    >
      <div
        className="Input"
      >
        <label
          className="Label"
        >
          quantity
        </label>
        <input
          className="InputElement ArmorShopListFormElement"
          disabled={false}
          max={1}
          min={0}
          name="quantity"
          onChange={[Function]}
          step={1}
          type="number"
          value={1}
        />
      </div>
      <button
        className="ArmorShopAddToCartButton"
        onClick={[Function]}
        type="submit"
      >
        Add to Cart
      </button>
    </form>
  </div>
</li>
`;

exports[`<ArmorShopItem /> should render with props passed in 1`] = `
<li
  className="ArmorShopListItem"
>
  <div
    className="ArmorShopListItemContainer"
  >
    <p>
      <strong>
        Name
      </strong>
      : 
      Legatron
    </p>
    <p>
      <strong>
        Type
      </strong>
      : 
      leg
    </p>
    <p>
      <strong>
        Description
      </strong>
      : 
      Leg Armor that is basic
    </p>
    <p>
      <strong>
        Company
      </strong>
      : 
      orian labs
    </p>
    <p>
      <strong>
        Quality
      </strong>
      : 
      low
    </p>
    <p>
      <strong>
        Protection
      </strong>
      : 
      low
    </p>
    <p>
      <strong>
        Shield
      </strong>
      : 
      0
    </p>
    <p>
      <strong>
        Discount
      </strong>
      : 
      0
    </p>
    <p>
      <strong>
        Cost
      </strong>
      : 
      100
    </p>
    <small>
      Created By: 
      Freddy
    </small>
  </div>
  <div
    className="ArmorShopListItemContainer"
  >
    <form
      className="ArmorShopListForm"
      onSubmit={[Function]}
    >
      <div
        className="Input"
      >
        <label
          className="Label"
        >
          quantity
        </label>
        <input
          className="InputElement ArmorShopListFormElement"
          disabled={true}
          max={0}
          min={0}
          name="quantity"
          onChange={[Function]}
          step={1}
          type="number"
          value={0}
        />
      </div>
      <button
        className="ArmorShopAddToCartButton ArmorShopAddToCartButtonDisabled"
        disabled={true}
        onClick={[Function]}
        type="submit"
      >
        Out of Stock
      </button>
    </form>
  </div>
</li>
`;
